cpp/ql/src/Best Practices/Unused Entities/UnusedLocals.ql

```diff
@@ -57,5 +57,10 @@ where
   not declarationHasSideEffects(v) and
   not exists(AsmStmt s | f = s.getEnclosingFunction()) and
   not v.getAnAttribute().getName() = "unused" and
-  not any(ErrorExpr e).getEnclosingFunction() = f // unextracted expr likely used `v`
+  not any(ErrorExpr e).getEnclosingFunction() = f and // unextracted expr may use `v`
+  not exists(Literal l |
+    l.getEnclosingFunction() = f and
+    not exists(l.getValue())
+  ) and
+  not any(ConditionDeclExpr cde).getEnclosingFunction() = f
```

Comment:
Please add some comments that'll help us know when to delete these extra conditions in the future.

Reply:
Done.  The details aren't really fresh in my mind any more, but at least its documented that they should at some point be removed.